src/server.ts

```diff
@@ -120,20 +172,83 @@ app.patch(
   }
 );
 
-app.delete(`/movies/:id`, requestLogger, (req: Request, res: Response) => {
+app.delete('/movies/:id', requestLogger, (req: Request, res: Response) => {
   const id = req.params.id;
   if (movies.find((m) => m.id === id)) {
     movies = movies.filter((m) => m.id !== id);
 
     res.send(`movie with id ${id} deleted`);
   } else {
-    res.statusCode = 404;
-    res.send('Invalid movie ID.');
+    res.status(404).json({ error: 'Invalid movie ID.' });
   }
 });
 
 app.use(errorHandler);
 
+const authRouter = express.Router();
+
+authRouter.post('/registration', (req: Request, res: Response) => {
+  const { name, pass } = req.body;
+  const user = db.find((user) => user.name === name);
+  if (user) {
+    res.status(409).json({ error: `User with name ${name} exists` });
+  } else {
+    db.push({
+      name: name,
+      pass: pass,
+      role: 'user',
+      favMovies: [],
+    });
+
+    res.status(201).json({ message: `User ${name} registered` });
+  }
+});
+
+authRouter.post('/login', (req: Request, res: Response) => {
+  const { name, pass } = req.body;
+  const user = db.find((user) => user.name === name && user.pass == pass);
```

Comment:
Saving passwords to DB without any encryption is very dangerous, in case you lost your data or someone gets access - all passwords will be revealed.
Try to avoid saving plain passwords.

Reply:
Added argon2 for pass hashing